src/traces/bar/defaults.js

```diff
@@ -33,7 +155,23 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     coerce('base');
     coerce('offset');
     coerce('width');
-    coerce('text');
+
+    coerceArray('text', coerceString);
+
+    var textPosition = coerceArray('textposition', coerceEnumerated);
+
+    var hasBoth = Array.isArray(textPosition) || textPosition === 'auto',
+        hasInside = hasBoth || textPosition === 'inside',
+        hasOutside = hasBoth || textPosition === 'outside';
+    if(hasInside || hasOutside) {
+        var textFont = coerceArray('textfont', coerceFont, layout.font);
+        if(hasInside) {
+            coerceArray('insidetextfont', coerceFont, textFont, layout.font);
+        }
+        if(hasOutside) {
+            coerceArray('outsidetextfont', coerceFont, textFont, layout.font);
+        }
+    }
```

Comment:
@etpinard I don't understand the phrase `"to make Plots.supplyDefaults scale ~ number of attribute"`? Could you give me an example?